pkg/apis/clusterregistry/v1alpha1/types.go

```diff
@@ -167,3 +171,87 @@ type ClusterCondition struct {
 	// +optional
 	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCredentials combines a Cluster record with a set of credentials
+// (stored in a kubeconfig) that can be used to access the cluster. The status
+// of ClusterCredentials reflects whether those credentials can be used - from
+// within the cluster hosting the ClusterCredentials resource - to access the
+// cluster's healthz endpoint.
+//
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=clustercredentials
+// +kubebuilder:subresource:status
+type ClusterCredentials struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterCredentialsSpec   `json:"spec,omitempty"`
+	Status ClusterCredentialsStatus `json:"status,omitempty"`
+}
+
+// ClusterCredentialsSpec defines the desired state of a ClusterCredentials.
+type ClusterCredentialsSpec struct {
+	// ClusterRef is a reference to the Cluster to access.
+	ClusterRef v1.LocalObjectReference `json:"clusterRef,omitempty"`
+
+	// SecretRef is a reference to the secret in the namespace that contains a
+	// kubeconfig with credentials access the referenced cluster. The secret
+	// must have a key named "kubeconfig" that contains a valid kubeconfig.
+	//
+	// TODO: explain required kubeconfig fields
+	// TODO: explain kubeconfig version
+	//
+	// This can be left empty if the cluster allows insecure access.
+	// +optional
+	SecretRef *apiv1.LocalObjectReference `json:"secretRef,omitempty"`
+}
+
+// ClusterCredentialsStatus contains information about the current status of a
+// ClusterCredentials.
+type ClusterCredentialsStatus struct {
```

Comment:
Note: I removed `availabilityZone` and `region` from the status of `FederatedCluster` when I used that as inspiration for this type. Do we think those concepts should be present in this API?

Reply:
IMHO, labels should be good enough like the labels on node. `failure-domain.beta.kubernetes.io/region` and `failure-domain.beta.kubernetes.io/zone`. we could use the same labels on cluster object.